src/hotspot/share/gc/serial/serialHeap.hpp

```diff
@@ -95,6 +95,13 @@ class SerialHeap : public CollectedHeap {
   GCMemoryManager* _young_manager;
   GCMemoryManager* _old_manager;
 
+  // Indicate whether heap is almost or approaching full.
+  // Usually, there is some memory headroom for application/gc to run properly.
+  // However, in extreme cases, e.g. young-gen is non-empty after a full gc, we
+  // will attempt some uncommon measures, e.g. alllocating small objs in
+  // old-gen.
+  bool _is_heap_almost_full;
```

Comment:
I'm not to fond of the `heap` part of the name since it lives in `SerialHeap`, but just calling it `_is_almost_full` will look strange since we also have the parameter `full` where it is used. 

What do you think about calling it something like `_allow_old_gen_allocation` instead?

Reply:
This field is meant to convey what the situation is, and how to handle this situation is decided in callers with some context.

Naming it `_allow_old_gen_allocation` would mix up what it is and what to do with that.